Update npm packages

mob8607 opened the pull request:
## Pull request
Updates outdated npm packages.

Currently **not** updated:
- Typescript from 4 to 5
- storybook from 6.5 to 7.6
 
### Browser testing
- http://localhost:5174/styleguide
- http://localhost:6006/?path=/docs/example-introduction--page
 
### Checklist
- [ ] I merged the current development branch (before testing)
- [ ] Added JSDoc and styleguide demo
- [ ] Tested all links in project relevant browsers
- [ ] Tested all links in different screen sizes
- [ ] Did run automated tests and linters
- [ ] Did assign ticket
- [ ] Double checked target branch
 
## Review/Test checklist
- [x] Did review code and documentation
- [ ] Tested all links in project relevant browsers
- [ ] Tested all links in different screen sizes
- [ ] Did check accessibility (Wave, only errors)
- [ ] Re-assign ticket to developer

patric-eberle commented on src/stories/components/c-modal.stories.js:
```diff
@@ -22,61 +22,73 @@ export default {
+    content: faker.lorem.paragraph(),
   },
 };
+const template = `
```
todo: please use a different name, since we already have `Template`.

benjamin-joham replied:
`template` is the default property name and also reflects the `<template>` part of the Vue SFC.
https://storybook.js.org/docs/vue/writing-stories/introduction#stories-for-two-or-more-components

I would rather go ahead and use a different name for the `const Template` on line 55.

patric-eberle commented on tests/unit/specs/plugins/vue-bem/testingEntitys.ts:
```diff
@@ -8,7 +8,17 @@ export const delimiters = {
+  mixin?: string;
+}
+
+interface Blocks {
```
todo: please use `type` instead. You can still extend this by using `type Foo = Blocks & {}`. `interface` should only be used in libraries, because they are shared on a global scope.

patric-eberle commented on tests/unit/specs/plugins/vue-bem/testingEntitys.ts:
```diff
@@ -25,7 +35,11 @@ export const blocks = {
 };
 
-export const elements = {
+interface Elements extends Blocks {
```
todo: please use a `type` instead`.

patric-eberle commented on tests/unit/specs/plugins/vue-bem/testingEntitys.ts:
```diff
@@ -48,7 +62,11 @@ export const elements = {
 };
 
-export const delimitersTest = {
+interface DelimitersTest {
```
todo: please use a `type` instead.

patric-eberle commented on tsconfig.json:
```diff
@@ -23,7 +23,8 @@
     "typeRoots": [
       "./src/types",
-      "./node_modules/@types/"
+      "./node_modules/@types/",
+      "./node_modules/"
```
suggestion: I would recommend to remove `node_modules/@types` in this case.

mob8607 replied:
That doesn't work as then he wont find the rest of the type definitions anymore.

patric-eberle reviewed (changes requested):
Browser review

- [x] A new major storybook version is available.
<img width="766" alt="grafik" src="https://github.com/valantic/vue-template/assets/10571945/643e89f4-8063-467f-9150-383b5698e9c3">

- [x] There is an error about a storybook add-on when running storybook:
<img width="731" alt="grafik" src="https://github.com/valantic/vue-template/assets/10571945/d528432e-01b2-4f4c-b7f2-6dd7cfedf0e3">

- [ ] There are two errors about static vs. dynamic imports when building. I think the one about `api` can be resolved by always do a static import. I think the other one can not be resolved.
<img width="802" alt="grafik" src="https://github.com/valantic/vue-template/assets/10571945/09d9746f-f337-43cf-8c70-247d3370b28a">

mob8607 commented:
> * There are two errors about static vs. dynamic imports when building. I think the one about `api` can be resolved by always do a static import. I think the other one can not be resolved.

I could not fix this without introducing another problem either in storybook or in the tests. I will for now leave it as it is. But we should investigate this a bit more, because to me it looks like we have a setup issue here.

patric-eberle commented:
@mob8607 this PR is still open. Do you have the possibility to check your questions with someone else? Not sure, when I will be ready to check your questions with you...

mob8607 commented:
@patric-eberle @benjamin-joham 
I updated the dependencies again. This time the stylelint version is updated (That's why i needed the commit for the valantic-styleguide dependency). This means that we now should integrate prettier in this release, because some of our rules are now dropped from stylelint.

patric-eberle commented:
> > * There are two errors about static vs. dynamic imports when building. I think the one about `api` can be resolved by always do a static import. I think the other one can not be resolved.
> 
> I could not fix this without introducing another problem either in storybook or in the tests. I will for now leave it as it is. But we should investigate this a bit more, because to me it looks like we have a setup issue here.

@mob8607 can you please create a ticket about what needs to be checked? I currently can't remember the whole context of this while you might do ;)

patric-eberle commented on src/elements/e-picture.vue:
```diff
@@ -119,12 +119,11 @@
        */
       loading: {
-        type: String,
+        type: String as PropType<'lazy' | 'eager'>,
```
Nice change 👍

patric-eberle reviewed (changes requested):
Added inputs about some package versions. Please double check.

- [ ] There is an endless list of warnings when running `npm run test`. I think this could be caused by the invalid vite version: `[eslint-plugin-import:resolver:vite] Cannot use import statement outside a module`
- [x] `npm run storybook` shows the following warning. Can we resolve this?
<img width="1165" alt="grafik" src="https://github.com/valantic/vue-template/assets/10571945/2073587a-ba6d-4d05-9b46-0ce05200d3d1">

patric-eberle commented:
@mob8607 can you please resolve the merge conflicts before I check this again. THX.

mob8607 commented:
@patric-eberle It should now be up to date and ready

patric-eberle commented:
@mob8607 changes look good to me. Thank you for your patience.  I was not able to force the `lint-staged` issue on my machine. Can you also document in this ticket, why you had to go back to vite 4? https://github.com/valantic/vue-template/issues/350

THX.
